### PR TITLE
Only disable quotas on the primary cluster

### PR DESCRIPTION
**Pull request by raskchanky**
https://hashicorp.atlassian.net/browse/VAULT-22381

tl;dr - setting up a rate limit quota inside a namespace and then deleting the namespace when perf replication is enabled causes the PR secondary to error when doing sync invalidation on the mount table and stop replication because it's trying to delete the quota from the PR secondary storage, which of course is read-only for quotas.

Making this change kept replication up and running and I verified that after the namespace was deleted on the primary, the associated rate limit was also gone from the secondary.

Although this change is to a CE file, it seems worth maybe opening this PR on ENT as well, just to let CI run?

### CODE REVIEW

**Comment by miagilepner** [comment]
> Although this change is to a CE file, it seems worth maybe opening this PR on ENT as well, just to let CI run?

I think that's a good idea. We could also add an ent test there for this scenario.

**Comment by raskchanky** [comment]
ENT PR with a test https://github.com/hashicorp/vault-enterprise/pull/5442

**Review by mpalmi — APPROVED** [review]
